### DISCUSSION TITLE
Why deprecating Dockerless installs?

### DISCUSSION THREAD

**Discussion by dani3l0** [wg-easy/wg-easy#1432] [Q&A]
I like the simplicity of UI, but I can't understand why we are forced to use Docker.
I already have a bare-metal Wireguard setup which is quite important to me. Bare-metal installations are more reliable imo, as Docker introduces some unnecessary layers of complexity like networking or isolation.

According to #1377:

> _Dockerless has some issues and isn’t widely used._
> _Removing it removes maintenance burden_

Really? Are those the reasons to drop simple bare-metal deployments?

When I do `npm i`, `npm ci` and `node server.js`, I am getting this response:

```
{
  "statusCode": 404,
  "statusMessage": "Cannot find static asset /",
  "stack": []
}
```

And the question is, is it still possible to run without Docker? If we are forced to use Docker, why?

**Comment by TheRealC0unt** [comment]
Yes, 100% agreement. I'm running both pihole and wg-easy on an old rpi3b, which is absolutely enough to access my home net from everywhere. I dislike the additional complexity docker will impose on me and the hassle this will cause in connection with the bare metal pihole DNS on the same machine.
KISS is totally preferable for crucial components like DNS and WireGuard.

**Comment by alex3025** [comment]
Yeah, I was running Dockerless wg-easy inside a Proxmox LXC... now I should install docker to run just a single container??

**Reply by divadwong68** [reply]
I've used this script for my LXC containers.
https://github.com/Nyr/wireguard-install/blob/master/wireguard-install.sh

**Comment by kaaax0815** [comment]
We conducted Poll https://github.com/wg-easy/wg-easy/discussions/1360 (arguably a little brief) because of issues like https://github.com/wg-easy/wg-easy/discussions/1382 which we were unable to reliably test.
Docker offers a way to hide the build steps from users, aligning with the project’s goal of simplifying the process.
With #1333, we aim to significantly enhance DX and UX, albeit at the cost of more complex build steps, which would complicate the documentation.

Therefore, we have decided to deprecate and remove Docker support. However, you can still set everything up manually. Following the instructions in https://github.com/wg-easy/wg-easy/wiki/%5BDeprecated-and-will-be-useless-with-version-15.0.0%5D-Using-WireGuard-Easy-without-Docker should work for version 14 at least.

For simpler environments, other solutions may be more suitable.

**Reply by dani3l0** [reply]
Thanks for answer, but can poll with just 4 votes be considered valid? Anyways, according to [Dockerfile](https://github.com/wg-easy/wg-easy/blob/feat-ui/Dockerfile) on a new branch `feat-ui`, setup is kinda more complicated than a standard npm project :thinking:, but still not such complicated to drop dockerless installs entirely.
I've tried to install it bare-metal on a plain debian VM, and ran just `pnpm install`, `pnpm build`, `cp .output/* .` and `sudo node server/index.mjs`. Everything looks ok, but hardcoding important parameters in code isn't a good practice (like [here](https://github.com/wg-easy/wg-easy/pull/1377/commits/a5260180bd8b4c1d8bc91e0ad056bc5f9ed56b76) or [here](https://github.com/wg-easy/wg-easy/pull/1333/commits/415e56c8252e780d5d206f53745fb27170ce286d#diff-96b5af49dc4ed02889d27c51fef75cd40f179a5dc00af811327c7a7d790f0e72)). Would be cool to have an option to set custom config paths or interface names

**Reply by kaaax0815** [reply]
I'll look into it. And maybe create a unofficial docs entry for it. And make Paths and such dynamic again

**Reply by abcd678** [reply]
Appreciate you @dani3l0 for bringing this to attention and @kaaax0815 for looking into it, I share the sentiment shared by others in this thread.
